### ISSUE TITLE
ttnn.group_norm accuracy issues on large tensors

### ISSUE THREAD

**Issue by djstefanovicTT** [tenstorrent/tt-metal#39553]
When running group norm on large tensors ([1,1280,720,256] in N H W C) pcc issues occur. Those problems are not in the entire resulting tensor, but in some columns along channel dimension. Values in these columns far exceed expected results. 
Example:
```
golden first 8=tensor([ 2.1194,  0.7311, -1.7896, -1.4202, -1.4262,  0.7760, -1.2356,  0.4882])
          y first 8=tensor([ 2.0156,  0.7383, -1.7344, -1.3359, -1.4062,  0.7461, -1.2344,  0.4863])
golden last 8=tensor([ 0.3884, -0.5064, -0.4964, -0.3148,  2.5813,  0.8681,  0.4419, -0.0662])
          y last 8=tensor([-2.0291e+21,  7.7822e+19, -6.4102e+20, -3.2282e+21, -3.1728e+21, -2.4719e+21, -1.6602e+20,  3.9199e+20])
```

Python script to reproduce and compare ttnn to torch:
```
import numpy as np
import torch
import ttnn


def pcc(a: torch.Tensor, b: torch.Tensor) -> float:
    a_np = a.float().reshape(-1).cpu().numpy()
    b_np = b.float().reshape(-1).cpu().numpy()
    return float(np.corrcoef(a_np, b_np)[0, 1])


def main():
    torch.manual_seed(0)
    device = ttnn.open_device(device_id=0)

    try:
        shape = (1, 1, 1280 * 720, 256)  # [N, 1, H*W, C]
        num_groups = 32
        eps = 1e-5

        x = torch.randn(shape, dtype=torch.bfloat16)
        weight = torch.randn((shape[-1],), dtype=torch.bfloat16)
        bias = torch.randn((shape[-1],), dtype=torch.bfloat16)
        c = shape[-1]
        weight_4d = weight.reshape(1, 1, c // 32, 32)
        bias_4d = bias.reshape(1, 1, c // 32, 32)

        # GroupNorm golden: convert [N,1,H*W,C] -> [N,C,1,H*W], apply GN, convert back.
        x_nchw = x.permute(0, 3, 1, 2).float()
        golden = torch.nn.functional.group_norm(
            x_nchw, num_groups=num_groups, weight=weight.float(), bias=bias.float(), eps=eps
        ).permute(0, 2, 3, 1)

        x_tt = ttnn.from_torch(x, device=device, layout=ttnn.TILE_LAYOUT)
        w_tt = ttnn.from_torch(weight_4d, device=device, layout=ttnn.ROW_MAJOR_LAYOUT)
        b_tt = ttnn.from_torch(bias_4d, device=device, layout=ttnn.ROW_MAJOR_LAYOUT)

        sharded_mem_config, grid_size = ttnn.determine_expected_group_norm_sharded_config_and_grid_size(
            device=device,
            num_channels=c,
            num_groups=num_groups,
            input_nhw=1280 * 720,
            is_height_sharded=False,
            is_row_major=False,
        )
        print(f"SMC {sharded_mem_config} GS {grid_size}")
        y_tt = ttnn.group_norm(
            x_tt,
            num_groups=num_groups,
            epsilon=eps,
            weight=w_tt,
            bias=b_tt,
            core_grid=grid_size,
            inplace=False,
            num_out_blocks=-1,
        )
        y = ttnn.to_torch(y_tt)

        pcc_val = pcc(golden, y)
        print(f"shape={shape}, num_groups={num_groups}, pcc={pcc_val:.6f}")

        preview = 8
        golden_flat = golden.flatten()
        y_flat = y.flatten()
        print(f"golden first {preview}={golden_flat[:preview]}")
        print(f"golden last {preview}={golden_flat[-preview:]}")
        print(f"y first {preview}={y_flat[:preview]}")
        print(f"y last {preview}={y_flat[-preview:]}")

    finally:
        ttnn.close_device(device)


if __name__ == "__main__":
    main()
```
